### ISSUE TITLE
CuDNN error: CUDNN_STATUS_EXECUTION_FAILED

### ISSUE THREAD

**Issue by aimhabo** [pytorch/pytorch#17971]
## 🐛 Bug
![2019-03-13 20-11-29](https://user-images.githubusercontent.com/12708080/54278734-2d791400-45ce-11e9-8926-64ffb9c5c2ed.png)
![2019-03-13 20-12-06](https://user-images.githubusercontent.com/12708080/54278743-34078b80-45ce-11e9-90ef-f2291ef9b95e.png)

## To Reproduce

Implementing such a network like this (or change from https://github.com/DrSleep/light-weight-refinenet  +  https://github.com/DrSleep/multi-task-refinenet)

![2019-03-13 20-39-56](https://user-images.githubusercontent.com/12708080/54279505-310d9a80-45d0-11e9-803a-0c7b5eeffb01.png)
![2019-03-13 20-39-35](https://user-images.githubusercontent.com/12708080/54279506-310d9a80-45d0-11e9-8429-bb7dfc595c70.png)
![2019-03-13 20-38-45](https://user-images.githubusercontent.com/12708080/54279508-31a63100-45d0-11e9-9463-71bd9e3ea21b.png)
![2019-03-13 20-38-34](https://user-images.githubusercontent.com/12708080/54279509-31a63100-45d0-11e9-87ee-bdd67041d22d.png)

and replace datasets.py to:
class Pad(object):
    """Pad image and mask to the desired size

    Args:
      size (int) : minimum length/width
      img_val (array) : image padding value
      msk_val (int) : mask padding value

    """
    def __init__(self, size, img_val, msk_val, dpt_val=None):
        self.size = size
        self.img_val = img_val
        self.msk_val = msk_val
        if dpt_val is None :
            self.dpt_val = msk_val
        else:
            self.dpt_val = dpt_val

    def __call__(self, sample):
        image, mask, depth = sample['image'], sample['mask'], sample['depth']
        h, w = image.shape[:2]
        h_pad = int(np.clip(((self.size - h) + 1)// 2, 0, 1e6))
        w_pad = int(np.clip(((self.size - w) + 1)// 2, 0, 1e6))
        pad = ((h_pad, h_pad), (w_pad, w_pad))
        image = np.stack([np.pad(image[:,:,c], pad,
                         mode='constant',
                         constant_values=self.img_val[c]) for c in range(3)], axis=2)
        mask = np.pad(mask, pad, mode='constant', constant_values=self.msk_val)
        depth = np.pad(depth, pad, mode='constant', constant_values=self.dpt_val)
        return {'image': image, 'mask': mask, 'depth': depth}

class RandomCrop(object):
    """Crop randomly the image in a sample.

    Args:
        output_size (tuple or int): Desired output size. If int, square crop
            is made.
    """

    def __init__(self, crop_size):
        assert isinstance(crop_size, int)
        self.crop_size = crop_size
        if self.crop_size % 2 != 0:
            self.crop_size -= 1

    def __call__(self, sample):
        image, mask, depth = sample['image'], sample['mask'], sample['depth']
        h, w = image.shape[:2]
        new_h = min(h, self.crop_size)
        new_w = min(w, self.crop_size)
        top = np.random.randint(0, h - new_h + 1)
        left = np.random.randint(0, w - new_w + 1)
        image = image[top: top + new_h,
                        left: left + new_w]
        mask = mask[top: top + new_h,
                    left: left + new_w]
        depth = depth[top: top + new_h,
                    left: left + new_w]
        return {'image': image, 'mask': mask, 'depth': depth}

class ResizeShorterScale(object):
    """Resize shorter side to a given value and randomly scale."""

    def __init__(self, shorter_side, low_scale, high_scale):
        assert isinstance(shorter_side, int)
        self.shorter_side = shorter_side
        self.low_scale = low_scale
        self.high_scale = high_scale

    def __call__(self, sample):
        image, mask, depth = sample['image'], sample['mask'], sample['depth']
        min_side = min(image.shape[:2])
        scale = np.random.uniform(self.low_scale, self.high_scale)
        if min_side * scale < self.shorter_side:
            scale = (self.shorter_side * 1. / min_side)
        image = cv2.resize(image, None, fx=scale, fy=scale, interpolation=cv2.INTER_CUBIC)
        mask = cv2.resize(mask, None, fx=scale, fy=scale, interpolation=cv2.INTER_NEAREST)
        depth = cv2.resize(depth, None, fx=scale, fy=scale, interpolation=cv2.INTER_NEAREST)
        return {'image': image, 'mask': mask, 'depth': depth}

class RandomMirror(object):
    """Randomly flip the image and the mask"""

    def __init__(self):
        pass

    def __call__(self, sample):
        image, mask, depth = sample['image'], sample['mask'], sample['depth']
        do_mirror = np.random.randint(2)
        if do_mirror:
            image = cv2.flip(image, 1)
            mask = cv2.flip(mask, 1)
            depth = cv2.flip(depth, 1)
        return {'image': image, 'mask' : mask, 'depth': depth}

class Normalise(object):
    """Normalise a tensor image with mean and standard deviation.
    Given mean: (R, G, B) and std: (R, G, B),
    will normalise each channel of the torch.*Tensor, i.e.
    channel = (channel - mean) / std

    Args:
        mean (sequence): Sequence of means for R, G, B channels respecitvely.
        std (sequence): Sequence of standard deviations for R, G, B channels
            respecitvely.
    """

    def __init__(self, scale, mean, std):
        self.scale = scale
        self.mean = mean
        self.std = std

    def __call__(self, sample):
        image = sample['image']
        return {'image': (self.scale * image - self.mean) / self.std, 'mask' : sample['mask'], 'depth': sample['depth']}

class ToTensor(object):
    """Convert ndarrays in sample to Tensors."""

    def __call__(self, sample):
        image, mask, depth = sample['image'], sample['mask'], sample['depth']
        # swap color axis because
        # numpy image: H x W x C
        # torch image: C X H X W
        image = image.transpose((2, 0, 1))
        return {'image': torch.from_numpy(image),
                'mask': torch.from_numpy(mask),
                'depth': torch.from_numpy(depth)}

class NYUDataset(Dataset):
    """NYUv2-40"""

    def __init__(
        self, data_file, data_dir, transform_trn=None, transform_val=None
        ):
        """
        Args:
            data_file (string): Path to the data file with annotations.
            data_dir (string): Directory with all the images.
            transform_{trn, val} (callable, optional): Optional transform to be applied
                on a sample.
        """
        with open(data_file, 'rb') as f:
            datalist = f.readlines()
        self.datalist = [(i,l,m) for i,l,m in map(lambda x: x.decode('utf-8').strip('\n').split('\t'), datalist)]
        self.root_dir = data_dir
        self.transform_trn = transform_trn
        self.transform_val = transform_val
        self.stage = 'train'

    def set_stage(self, stage):
        self.stage = stage

    def __len__(self):
        return len(self.datalist)

    def __getitem__(self, idx):
        #img_name = os.path.join(self.root_dir, self.datalist[idx][0])
        #msk_name = os.path.join(self.root_dir, self.datalist[idx][1])
        img_name = self.datalist[idx][0]
        msk_name = self.datalist[idx][1]
        dpt_name = self.datalist[idx][2]
        def read_image(x):
            img_arr = np.array(Image.open(x))
            if len(img_arr.shape) == 2: # grayscale
                img_arr = np.tile(img_arr, [3, 1, 1]).transpose(1, 2, 0)
            return img_arr
        image = read_image(img_name)
        mask = np.array(Image.open(msk_name))
        depth = np.array(Image.open(dpt_name))
        if img_name != msk_name:
            assert len(mask.shape) == 2, 'Masks must be encoded without colourmap'
        sample = {'image': image, 'mask': mask, 'depth': depth}
        if self.stage == 'train':
            if self.transform_trn:
                sample = self.transform_trn(sample)
        elif self.stage == 'val':
            if self.transform_val:
                sample = self.transform_val(sample)
        return sample

## Environment

conda install --yes --file requirements.txt

# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: linux-64
atomicwrites=1.2.1=py36_0
attrs=18.2.0=py36h28b3542_0
backcall=0.1.0=py36_0
blas=1.0=mkl
ca-certificates=2018.03.07=0
certifi=2018.10.15=py36_0
cffi=1.11.5=py36he75722e_1
cloudpickle=0.6.1=py36_0
cudatoolkit=9.0=h13b8566_0
cudnn=7.1.2=cuda9.0_0
cycler=0.10.0=py36h93f1223_0
cython=0.29=py36he6710b0_0
cytoolz=0.9.0.1=py36h14c3975_1
dask-core=0.20.0=py36_0
dbus=1.13.2=h714fa37_1
decorator=4.3.0=py36_0
expat=2.2.6=he6710b0_0
fontconfig=2.13.0=h9420a91_0
freetype=2.9.1=h8a8886c_1
glib=2.56.2=hd408876_0
gst-plugins-base=1.14.0=hbbd80ab_1
gstreamer=1.14.0=hb453b48_1
icu=58.2=h9c2bf20_1
imageio=2.4.1=py36_0
intel-openmp=2019.0=118
ipython=7.1.1=py36h39e3cac_0
ipython_genutils=0.2.0=py36hb52b0d5_0
jedi=0.13.1=py36_0
jpeg=9b=h024ee3a_2
kiwisolver=1.0.1=py36hf484d3e_0
libedit=3.1.20170329=h6b74fdf_2
libffi=3.2.1=hd88cf55_4
libgcc-ng=8.2.0=hdf63c60_1
libgfortran-ng=7.3.0=hdf63c60_0
libpng=1.6.35=hbc83047_0
libstdcxx-ng=8.2.0=hdf63c60_1
libtiff=4.0.9=he85c1e1_2
libuuid=1.0.3=h1bed415_2
libxcb=1.13=h1bed415_1
libxml2=2.9.8=h26e45fe_1
markdown=2.6.8=py36_0
matplotlib=3.0.1=py36h5429711_0
mkl=2018.0.3=1
mkl_fft=1.0.6=py36h7dd41cf_0
mkl_random=1.0.1=py36h4414c95_1
more-itertools=4.3.0=py36_0
nccl=1.3.5=cuda9.0_0
ncurses=6.1=hf484d3e_0
networkx=2.2=py36_1
ninja=1.8.2=py36h6bb024c_1
numpy=1.15.3=py36h1d66e8a_0
numpy-base=1.15.3=py36h81de0dd_0
olefile=0.46=py36_0
openssl=1.0.2p=h14c3975_0
parso=0.3.1=py36_0
pcre=8.42=h439df22_0
pexpect=4.6.0=py36_0
pickleshare=0.7.5=py36_0
pillow=5.3.0=py36h34e0f95_0
pip=10.0.1=py36_0
pluggy=0.8.0=py36_0
prompt_toolkit=2.0.7=py36_0
ptyprocess=0.6.0=py36_0
py=1.7.0=py36_0
pycparser=2.19=py36_0
pydot=1.2.4=py36_0
pygments=2.2.0=py36h0d3125c_0
pyparsing=2.2.2=py36_0
pyqt=5.9.2=py36h05f1152_2
pytest=3.9.3=py36_0
pytest-runner=4.2=py36_0
python=3.6.6=h6e4f718_2
python-dateutil=2.7.5=py36_0
pytorch=0.4.1=py36_py35_py27__9.0.176_7.1.2_2
pytz=2018.7=py36_0
pywavelets=1.0.1=py36hdd07704_0
qt=5.9.6=h8703b6f_2
readline=7.0=h7b6447c_5
scikit-image=0.14.1=py36he6710b0_0
scipy=1.1.0=py36hfa4b5c9_1
setuptools=40.4.3=py36_0
sip=4.19.8=py36hf484d3e_0
six=1.11.0=py36_1
sqlite=3.25.2=h7b6447c_0
tk=8.6.8=hbc83047_0
toolz=0.9.0=py36_0
torchvision=0.2.1=py_2
tornado=5.1.1=py36h7b6447c_0
traitlets=4.3.2=py36_0
wcwidth=0.1.7=py36_0
wheel=0.32.2=py36_0
xz=5.2.4=h14c3975_4
zlib=1.2.11=ha838bed_2

```

 - PyTorch Version (e.g., 1.0):  0.4.1
 - OS (e.g., Linux): ubuntu 16.04.5
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): conda install --yes --file requirements.txt 
 - Python version: 3.6
 - CUDA/cuDNN version: 9.0/7.1.2
 - GPU models and configuration: GTX1070Ti
 - Any other relevant information:

**Comment by xinliweiyuan** [comment]
It seems the bug or problem of Pytorch 1.1. But why do you have such a problem in 0.4?

**Comment by xinliweiyuan** [comment]
However, I fixed the bug, but not from codes. I change the python to Conda python, and the problem is solved. So the codes still not compatible with the original python3. When installing from Conda, the Cuda toolkit will be installed, and it might be the solution.

blas-1.0-mkl.t 100% |##################################################| Time: 0:00:00   8.66 MB/s
conda-env-2.6. 100% |##################################################| Time: 0:00:00 710.85 kB/s
cudatoolkit-10 100% |##################################################| Time: 0:43:28 152.74 kB/s
ninja-1.7.2-0. 100% |##################################################| Time: 0:00:00   4.63 MB/s
pytorch-1.1.0- 100% |##################################################| Time: 0:46:31 170.90 kB/s
torchvision-0. 100% |##################################################| Time: 0:00:00  60.07 kB/s
conda-4.3.30-p 100% |#####################

@jeffreyksmithjr @vishwakftw

**Comment by moyans** [comment]
if you runing pytorch in docker,  you shuld know that: https://github.com/NVIDIA/tacotron2/issues/109
